app/Http/Controllers/NewsController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Http\Controllers;
+use App\Models\News;
+use Illuminate\Http\Request;
+use App\Http\Requests\NewsRequest;  
+
+class NewsController extends Controller
+{
+
+    protected $newsModel;
+
+    public function __construct(News $newsModel)
+    {
+        $this->newsModel = $newsModel;
+    }
+
+    public function index(Request $request)
+    {
+        $query = $this->newsModel->query(); // Tạo truy vấn
+
+        // Xử lý tìm kiếm theo tên (name) nếu có
+        if ($request->has('name')) {
+            $name = $request->input('name');
+            $query->where('name', 'like', "%$name%");
+        }
+
+        $news = $query->paginate(5); // Thực hiện truy vấn phân trang
+    
+        return view('news.index', [
+            'news' => $news
+        ]);
+    }
+    
+
+    /**
+     * Show the form for creating a new resource.
+     */
+    public function create()
+    {
+        return view('news.form');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     */
+    public function store(NewsRequest $request)
+    {
+        $data = $request->validated(); // Lấy dữ liệu đã được kiểm tra và hợp lệ từ request
```

Comment:
Ko cần dùng khai báo thêm biến data

` $this->newsModel->create($request->validated());`

Reply:
fixed